SPARQLRepositoryTest fails when targeting AWS node (vs localhost)

Test com.marklogic.semantics.rdf4j.SPARQLRepositoryTest on develop branch fails when using AWS node:

COMMAND: ./gradlew :marklogic-rdf4j:test
.....
com.marklogic.semantics.rdf4j.SPARQLRepositoryTest > testSPARQLRepositoryWithMarkLogic STARTED

com.marklogic.semantics.rdf4j.SPARQLRepositoryTest > testSPARQLRepositoryWithMarkLogic FAILED
org.eclipse.rdf4j.repository.RepositoryException at SPARQLRepositoryTest.java:64
Caused by: org.eclipse.rdf4j.query.UpdateExecutionException at SPARQLRepositoryTest.java:64
Caused by: org.eclipse.rdf4j.http.protocol.UnauthorizedException at SPARQLRepositoryTest.java:64

--- a/marklogic-rdf4j/src/main/java/com/marklogic/semantics/rdf4j/utils/Util.java
+++ b/marklogic-rdf4j/src/main/java/com/marklogic/semantics/rdf4j/utils/Util.java
@@ -87,7 +87,7 @@ public class Util {
      */
     public Value skolemize(Value s) {
         if (s instanceof org.eclipse.rdf4j.model.BNode) {
-            return SimpleValueFactory.getInstance().createIRI("http://marklogic.com/semantics/blank/" + s.toString());
+            return SimpleValueFactory.getInstance().createIRI("http://marklogic.com/semantics/blank/" + s.stringValue());
         } else {
             return s;
         }

--- a/marklogic-rdf4j/src/test/java/com/marklogic/semantics/rdf4j/MarkLogicRepositoryConnectionTest.java
+++ b/marklogic-rdf4j/src/test/java/com/marklogic/semantics/rdf4j/MarkLogicRepositoryConnectionTest.java
@@ -667,6 +667,22 @@ public class MarkLogicRepositoryConnectionTest extends Rdf4jTestBase {
         Assert.assertTrue(conn.hasStatement(null, null, copyOfAliceName, false));
     }
 
+    // https://github.com/marklogic/marklogic-rdf4j/issues/39
+    @Test
+    public void testBNodeSkolmization() throws Exception
+    {
+        ValueFactory vf = conn.getValueFactory();
+        Resource context1 = vf.createIRI("http://marklogic.com/test/context1");
+        IRI alice = vf.createIRI("http://example.org/people/alice");
+        IRI name = vf.createIRI("http://example.org/ontology/name");
+        BNode alicesName = vf.createBNode();
+        Statement st1 = vf.createStatement(alice, name, alicesName);
+        conn.add(st1, context1);
+        final GraphQueryResult evaluate = conn.prepareGraphQuery("DESCRIBE <http://example.org/people/alice>").evaluate();
+
+        Assert.assertTrue(evaluate.hasNext());
+    }
+
     // https://github.com/marklogic/marklogic-sesame/issues/363
     @Test
     public void testMultiContextDelete()

--- a/marklogic-rdf4j/src/test/java/com/marklogic/semantics/rdf4j/SPARQLRepositoryTest.java
+++ b/marklogic-rdf4j/src/test/java/com/marklogic/semantics/rdf4j/SPARQLRepositoryTest.java
@@ -49,7 +49,7 @@ public class SPARQLRepositoryTest extends Rdf4jTestBase {
             throws Exception {
         logger.debug("setting up test");
         SPARQLRepository sr = new SPARQLRepository("http://"+host+":"+port+"/v1/graphs/sparql");
-        sr.setUsernameAndPassword("admin","admin");
+        sr.setUsernameAndPassword(adminUser, adminPassword);
         sr.initialize();
         f = sr.getValueFactory();
         conn = (SPARQLConnection) sr.getConnection();

